include/linker/utils.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_LINKER_UTILS_H_
+#define ZEPHYR_INCLUDE_LINKER_UTILS_H_
+
+#include <stdbool.h>
+
+/**
+ * @brief Check if address is in read only section.
+ *
+ * @param addr Address.
+ *
+ * @return True if address identified within read only section.
+ */
+static inline bool lnkr_is_in_rodata(const void *addr)
```

Comment:
Might want to limit the scope here to a "is_string_literal()" predicate, since AFAICT that's what it's being used to detect.  There are platforms where it's convenient to link data directly into .text, where there are multiple regions for immutable data (the DSP is moving in this direction with __imr) etc... The toolchain behavior with string literals isn't well specified or portable either, but it's likely to be more easily pinned down in the face of weird platform linkage.

Reply:
This corresponds to the python script to extract strings from ELF file, where strings are extracted from inside `.rodata`.

Note that this will also be used for cbprintf once the bug on SPARC is fixed there (https://github.com/zephyrproject-rtos/zephyr/issues/42208).

Reply:
Done PR for cbprintf #42577

Reply:
Thank you. Will rebase.

Reply:
@dcpleung Could you extract this one into separate PR? That could get in earlier. While working on cbprintf extension i could use it there.
Also i would not abbreviate `linker` to `lnkr`. Function name is not that long.

Reply:
Regarding @andyross comment, maybe you could add note to the description that variable may be in read only section/memory even though function return `false`.

Reply:
Submitted the linker commit @ https://github.com/zephyrproject-rtos/zephyr/pull/43168